### ISSUE TITLE
Lint/ToEnumArguments false negative for braced hash passed to keyword arguments

### ISSUE THREAD

**Issue by RedZapdos123** [rubocop/rubocop#15290]
## Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

## Issue type

False negative

## Cop name

`Lint/ToEnumArguments`

## Steps to reproduce the problem

1. Create a file `example.rb` with:

```ruby
def each_item(limit:)
  return to_enum(__method__, { limit: limit }) unless block_given?
end
```

2. Run:

```sh
bundle exec ruby exe/rubocop example.rb --only Lint/ToEnumArguments
```

3. RuboCop reports no offense.

4. But the enumerator re-invokes the method with a positional hash, which raises:

```ruby
def each_item(limit:)
  return to_enum(__method__, { limit: limit }) unless block_given?
end

enum = each_item(limit: 1)
enum.each {}
```

```text
ArgumentError
wrong number of arguments (given 1, expected 0; required keyword: limit)
```

## Expected behavior

RuboCop should register an offense.

`to_enum(__method__, { limit: limit })` passes a positional hash, not proper keyword argument forwarding for a method with `limit:`.

## Actual behavior

RuboCop does not register an offense.

The current cop accepts the braced hash as if it correctly satisfied the keyword argument.

## RuboCop configuration

No special configuration.

## RuboCop version

```shell
$ bundle exec ruby exe/rubocop -V
1.87.0 (using Parser 3.3.11.1, rubocop-ast 1.49.1, analyzing as Ruby 2.7, running on ruby 3.2.3) +Rubydex [x86_64-linux-gnu]
  - rubocop-performance 1.26.1
  - rubocop-rake 0.7.1
  - rubocop-rspec 3.10.2
```

**Comment by RedZapdos123** [comment]
Opened a PR to resolve this issue.
